Group detail - Permission tab UX changes

himdel opened the pull request:
Fixes https://issues.redhat.com/browse/AAH-364

This contains multiple UX changes to the Group detail screen, mostly in the Permissions tab:

* moved the Edit, Save and Cancel buttons from page header to tab header, as these are only related to the tab
* keeps the Delete button as is, but disables it when editing a group, and no longer hides it on the users tab
* makes the tabs appear disabled (cursor + no action on click) when editing Permissions
* sorts the list of available permissions in the Permission selects (note this makes the order unstable between translations once we have i18n)
* use plain text styling (remove gray blocks behind input text) unless in edit mode (this is from https://issues.redhat.com/browse/AAH-360 but consistency)

Cc @ZitaNemeckova, @sbuenafe-rh, this should be ready for review :)

---

Permissions tab in view mode (before and after):

![perm false before](https://user-images.githubusercontent.com/289743/110563428-36f8d080-8143-11eb-95b0-471967ff695c.png)
![perm false after](https://user-images.githubusercontent.com/289743/110563432-38c29400-8143-11eb-8426-b962fe75d554.png)


Permissions tab in edit mode (before and after):

![perm true before](https://user-images.githubusercontent.com/289743/110563486-4c6dfa80-8143-11eb-8bc7-f0d6bd3bc49a.png)
![perm true after](https://user-images.githubusercontent.com/289743/110563476-48da7380-8143-11eb-8747-8b81ee7a2e36.png)

(the tab header also switches to the `not-allowed` cursor when hovering in edit mode)


List of group permissions when editing (before and after):

![perm true2 before](https://user-images.githubusercontent.com/289743/110563589-73c4c780-8143-11eb-8a31-837f7e3b7a61.png)
![perm true2 after](https://user-images.githubusercontent.com/289743/110563597-76bfb800-8143-11eb-9304-c04d6d345347.png)


Users tab (before and after):

![user before](https://user-images.githubusercontent.com/289743/110563628-7cb59900-8143-11eb-8b24-61a38709e734.png)
![user after](https://user-images.githubusercontent.com/289743/110563636-7e7f5c80-8143-11eb-8a20-bb5870b8787c.png)

ZitaNemeckova commented on src/containers/group-management/group-detail.tsx:
```diff
@@ -299,7 +301,8 @@ class GroupDetail extends React.Component<RouteComponentProps, IState> {
                       twoWayMapper(value, Constants.HUMAN_PERMISSIONS),
-                    )}
+                    )
+                    .sort()}
```
Or sort it at `src/constants.tsx`. WDYT?

himdel replied:
Ah, good idea :+1: This would make the order stable accross translations, so I suppose it depends on that :)

ZitaNemeckova commented:
- Disabled tab has a text like "You must finish editing"
- Permissions in view mode should be labels
<img width="1371" alt="Screenshot 2021-03-10 at 10 55 54" src="https://user-images.githubusercontent.com/9210860/110611134-371ebd80-818f-11eb-905a-4bd907ec03b4.png">
@sbuenafe-rh WDYT?

LGTM code-wise 👍

sbuenafe-rh commented:
@himdel Looks good, but I would move the Save & Cancel to the bottom left (just to be consistent with the ReadMe Markdown editor)

himdel commented:
Updated to match the labels screenshot, thanks! :)
And added a tab title when disabled, except "you must" felt a bit strong, wdyt?..

![IMG_20210311_014041](https://user-images.githubusercontent.com/289743/110718400-d03aec00-8202-11eb-9c56-17dc2007ff5b.jpg)

And moved the Save & Cancel buttons down, and changed to primary+secondary (from default+link) as in the editor (is that what we want to settle on?);  leaving the Edit button as is..

![savecancel](https://user-images.githubusercontent.com/289743/110719212-59065780-8204-11eb-98f5-009338e9b450.png)

ZitaNemeckova reviewed (approved):
LGTM :+1:
